carma_cooperative_perception/src/msg_conversion.cpp

```diff
@@ -569,107 +569,113 @@ auto to_detection_list_msg(
+        quat_tf.setRPY(0, 0, remove_units(units::angle::radian_t{enu_yaw}));
+      }
 
-    detection.pose.pose.orientation = tf2::toMsg(quat_tf);
+      detection.pose.pose.orientation = tf2::toMsg(quat_tf);
 
-    const auto speed{Speed::from_msg(common_data.speed)};
-    detection.twist.twist.linear.x =
-      remove_units(units::velocity::meters_per_second_t{speed.speed});
+      const auto speed{Speed::from_msg(common_data.speed)};
+      detection.twist.twist.linear.x =
+        remove_units(units::velocity::meters_per_second_t{speed.speed});
 
-    if (!common_data.speed_z.unavailable){
-      const auto speed_z{Speed::from_msg(common_data.speed_z)};
-      detection.twist.twist.linear.z =
-        remove_units(units::velocity::meters_per_second_t{speed_z.speed});
-    }
-    else{
-      detection.twist.twist.linear.z = remove_units(units::velocity::meters_per_second_t{0.0});
-    }
+      if (!common_data.speed_z.unavailable){
+        const auto speed_z{Speed::from_msg(common_data.speed_z)};
+        detection.twist.twist.linear.z =
+          remove_units(units::velocity::meters_per_second_t{speed_z.speed});
+      }
+      else{
+        detection.twist.twist.linear.z = remove_units(units::velocity::meters_per_second_t{0.0});
+      }
 
-    // NOTE: common_data.accel_4_way.longitudinal, lateral, vert not supported
-    // and not needed at the moment for multiple object tracking algorithm
-    // Having non-zero yaw_rate value means available
-    if(static_cast<bool>(common_data.accel_4_way.yaw_rate)){
-      const auto accel_set{AccelerationSet4Way::from_msg(common_data.accel_4_way)};
-      detection.twist.twist.angular.z =
-        remove_units(units::angular_velocity::degrees_per_second_t{accel_set.yaw_rate});
-    }
-    else{
-      detection.twist.twist.angular.z = 0.0;
-    }
+      // NOTE: common_data.accel_4_way.longitudinal, lateral, vert not supported
+      // and not needed at the moment for multiple object tracking algorithm
+      // Having non-zero yaw_rate value means available
+      if(static_cast<bool>(common_data.accel_4_way.yaw_rate)){
+        const auto accel_set{AccelerationSet4Way::from_msg(common_data.accel_4_way)};
+        detection.twist.twist.angular.z =
+          remove_units(units::angular_velocity::degrees_per_second_t{accel_set.yaw_rate});
+      }
+      else{
+        detection.twist.twist.angular.z = 0.0;
+      }
 
-    convert_covariances(detection, common_data, conversion_adjustment);
+      convert_covariances(detection, common_data, conversion_adjustment);
 
-    convert_object_type(detection, common_data.obj_type);
+      convert_object_type(detection, common_data.obj_type);
 
-    detection_list.detections.push_back(std::move(detection));
+      detection_list.detections.push_back(std::move(detection));
+    }
   }
-
+  catch (...) {
+    RCLCPP_ERROR_STREAM(rclcpp::get_logger("sdsm_to_detection_list_node"), "Error converting SDSM to object, ignoring sdsm message.")
+  }
```

Comment:
If the issue is really only due to missing georeference, we should just add that guard and not catch all error like this.
The georeference guard would be on top of the sdsm_callback https://github.com/usdot-fhwa-stol/carma-platform/blob/fix-cp-sdsm-missing-georeference/carma_cooperative_perception/include/carma_cooperative_perception/sdsm_to_detection_list_component.hpp#L110